src/accessories/characteristics/currentPosition.ts

```diff
@@ -15,43 +16,10 @@ export class CurrentPositionCharacteristic extends TuyaWebCharacteristic {
   }
 
   public getRemoteValue(callback: CharacteristicGetCallback): void {
-    this.accessory
-      .getDeviceState()
-      .then((data) => {
-        this.debug("[GET] %s", data?.state);
-        this.updateValue(data, callback);
-      })
-      .catch(this.accessory.handleError("GET", callback));
+    this.updateValue({}, callback);
   }
 
   updateValue(data: DeviceState, callback?: CharacteristicGetCallback): void {
-    this.debug(`Updating value`, data);
-    if (!isNaN(Number(String(data?.state)))) {
-      //State is a number and probably 1, 2 or 3
-      const state = Number(data.state);
-      const stateValue = {
-        1: 100,
-        2: 50,
-        3: 0,
-      }[state];
-
-      this.accessory.setCharacteristic(
-        this.homekitCharacteristic,
-        stateValue,
-        !callback
-      );
-      callback && callback(null, stateValue);
-    } else if (["true", "false"].includes(String(data?.state).toLowerCase())) {
-      const stateValue = String(data.state).toLowerCase() === "true" ? 100 : 0;
-      this.accessory.setCharacteristic(
-        this.homekitCharacteristic,
-        stateValue,
-        !callback
-      );
-      callback && callback(null, stateValue);
-    } else {
-      callback &&
-        callback(new Error(`Unexpected state value provided: ${data?.state}`));
-    }
+    callback && callback(null, (<CoverAccessory>this.accessory).position);
```

Comment:
This will fail when `this.accessory` is an instance of `WindowAccessory` which also uses this characteristic